File: app/src/main/java/ru/systempla/weatherapp/di/module/LocationListenerModule.java
package ru.systempla.weatherapp.di.module;

import android.location.LocationListener;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.systempla.weatherapp.mvp.model.location.BlankLocationListener;

@Module
public class LocationListenerModule {

    @Singleton
    @Provides
    public LocationListener locationListener() {
        return new BlankLocationListener();
    }
}

File: app/src/main/java/ru/systempla/weatherapp/di/module/LocationListenerModule.kt
package ru.systempla.weatherapp.di.module

import android.location.LocationListener
import dagger.Module
import dagger.Provides
import ru.systempla.weatherapp.mvp.model.location.BlankLocationListener
import javax.inject.Singleton

@Module
class LocationListenerModule {
    @Singleton
    @Provides
    fun locationListener(): LocationListener = BlankLocationListener()
}